### PR TITLE
Normalize allowlist lookups to use lowercase non-checksummed authenticated address

### PR DESCRIPTION
**Pull request by pschork**
## Why are these changes needed?

This mitigates a recent incident where the allowlist contained a non-checksummed address for LayerN, but LayerN requests contained a checksummed address resulting in failed rateConfig lookup.

This change normalizes the addresses to lowercase when loading the allowlist rateConfig, and always lowers the authenticatedAccount when doing rate bucket lookup.

This approach allows the allowList to be specified as checksummed or non-checksummed address, as well as the dispersalRequest account to be specified as checksummed or non-checksummed address.

## Checks

- [x] I've made sure the lint is passing in this PR.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- [x] I've checked the new test coverage and the coverage percentage didn't drop.
- Testing Strategy
   - [x] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: disperser/apiserver/server.go [hunk 1]
```diff
@@ -326,6 +326,15 @@ func (s *DispersalServer) getAccountRate(origin, authenticatedAddress string, qu
 	// Check if the address is in the allowlist
 	if len(authenticatedAddress) > 0 {
 		quorumRates, ok := s.rateConfig.Allowlist[authenticatedAddress]
```
Comment (ian-shim) [inline disperser/apiserver/server.go:328]: how about we always make the keys in this map lowercase and compare with lowercased addresses? 
https://github.com/Layr-Labs/eigenda/blob/master/disperser/apiserver/config.go#L167-L170

Comment (pschork) [reply disperser/apiserver/server.go:328]: Thx for pointing this out. That's a way more systemic solution and avoids us having to normalize the allowlist configs.

File: disperser/apiserver/config.go [hunk 2]
```diff
@@ -164,9 +165,11 @@ func ReadAllowlistFromFile(f string) (Allowlist, error) {
 	}
 
 	for _, entry := range allowlistEntries {
-		rateInfoByQuorum, ok := allowlist[entry.Account]
+		// normalize to lowercase (non-checksummed) address
```
Comment (ian-shim) [inline disperser/apiserver/config.go:168]: nit: can we say this is either address or IP address? (not necessarily address)

File: disperser/apiserver/server.go [hunk 3]
```diff
@@ -325,6 +325,9 @@ func (s *DispersalServer) getAccountRate(origin, authenticatedAddress string, qu
 
 	// Check if the address is in the allowlist
 	if len(authenticatedAddress) > 0 {
+		// normalize to lowercase (non-checksummed) address
```
Comment (ian-shim) [inline disperser/apiserver/server.go:328]: same here

**Review by ian-shim — APPROVED** [review]
Thanks for fixing this!

**Review by bxue-l2 — APPROVED** [review]

**Pull request merged**